app/src/BarChart.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * Created by ayub on 11/10/19.
+ */
+import React from "react";
+import * as d3 from "d3";
+
+const colorScale = d3.scaleOrdinal(d3.schemeCategory10);
+const format = d3.format(",");
+
+class BarChart extends React.Component
+{
+    constructor(props){
+        super(props);
+        this.state = {
+            sort: true,
+            species: [...this.props.data.species].sort((a, b) => b.value - a.value),
+            xScale : d3
+                .scaleBand()
+                .range([0, this.props.width - this.props.left - this.props.right])
+                .domain([...this.props.data.species].sort((a, b) => b.value - a.value).map(d => d.name))
+                .padding(0.1),
+
+            yScale : d3
+                .scaleLinear()
+                .range([this.props.height - this.props.top - this.props.bottom, 0])
+                .domain([0, d3.max(this.props.data.species, d => d.value)])
+        }
+    }
+
+    setLocalState(){
+        this.setState({
+            species: [...this.props.data.species].sort((a, b) => b.value - a.value),
+            xScale : d3
+                .scaleBand()
+                .range([0, this.props.width - this.props.left - this.props.right])
+                .domain([...this.props.data.species].sort((a, b) => b.value - a.value).map(d => d.name))
+                .padding(0.1),
+
+            yScale : d3
+                .scaleLinear()
+                .range([this.props.height - this.props.top - this.props.bottom, 0])
+                .domain([0, d3.max(this.props.data.species, d => d.value)])
+        })
+    }
+
+    componentDidUpdate(prevProps) {
+        if(this.props.data.state !== prevProps.data.state || this.props.data.unit !== prevProps.data.unit
+            || this.props.data.year !== prevProps.data.year||this.props.data.species !== prevProps.data.species){
+            this.setLocalState();
+        }
+        this.drawChart();
+    }
+
+
+    plot(chart, width, height) {
+
+        // Define the div for the tooltip
+        var div = d3.select("body").append("div")
+            .attr("class", "tooltip")
+            .style("opacity", 0);
+
+        var state = this.props.data.state;
```

Comment:
was naming this `usState` to avoid confusion with the `state` name that gets used by React all over the place